packages/babel-plugin-bugfix-v8-static-class-fields-redefine-readonly/src/util.ts

```diff
@@ -0,0 +1,220 @@
+import type { NodePath, Visitor } from "@babel/traverse";
+import { types as t } from "@babel/core";
+
+function isNameOrLength(key: t.Node): boolean {
+  if (t.isIdentifier(key)) {
+    return key.name === "name" || key.name === "length";
+  }
+  if (t.isStringLiteral(key)) {
+    return key.value === "name" || key.value === "length";
+  }
+  return false;
+}
+
+function isStaticFieldWithValue(
+  node: t.Node,
+): node is t.ClassProperty | t.ClassPrivateProperty {
+  return (
+    (t.isClassProperty(node) || t.isClassPrivateProperty(node)) &&
+    node.static &&
+    !!node.value
+  );
+}
+
+const hasReferenceVisitor: Visitor<{ name: string; ref: () => void }> = {
+  ReferencedIdentifier(path, state) {
+    if (path.node.name === state.name) {
+      state.ref();
+      path.stop();
+    }
+  },
+  Scope(path, { name }) {
```

Comment:
Edge case: the visitor here should be merged with environment visitor.

Reply:
Could you provide a test case? Since this doesn't use `this` I am having trouble with finding an example :sweat_smile:

Reply:
Sure, maybe
```js
class C {
  static p = class {
    [C.name](C = 42) {}
  }
}
```

Reply:
I need to use `skipAndrequeueComputedKeysAndDecorators` from #14371, because I only want to skip the method if it declares `C` (`environmentVisitor` would always skip it).